api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -288,4 +288,23 @@
   {
 
   }
+
+  /**
+   * This class provides functionality to extend existing DAG. The type of extensions allowed are
+   *
+   * <li>Add a new operator</li>
+   * <li>Remove an existing operator</li>
+   * <li>Add an operator to stream</li>
+   * <li>Add an stream between existing operator and new opeartor</li>
+   */
+  interface DAGChangeSet extends DAG
```

Comment:
My idea was to use the same api to specify dag changes with
addition of new methods to change existing DAG. This also
allows running existing application as a subdag, for
example following application run three applications one after
another, individual applications can be run separately too

https://github.com/tushargosavi/apex-dynamic-scheduling/blob/master/src/main/java/com/datatorrent/wordcount/lindag/DagSchedulingApp.java#L35